Slider/Assets/Scripts/Misc/SimpleInteractableCutscene.cs

```diff
@@ -0,0 +1,200 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using UnityEngine;
+
+public class SimpleInteractableCutscene : MonoBehaviour, IInteractable
+{
+    [Tooltip("Optional Flag to set when cutscene starts")]
+    [SerializeField] private string cutsceneStartedFlag;
+    [Tooltip("Optional Flag to set when cutscene finishes")]
+    [SerializeField] private string cutsceneFinishedFlag;
+    [Tooltip("Array of all the characters in the cutscene")]
+    [SerializeField] protected NPC[] cutsceneCharacters;
+
+    private const float DEFAULT_TIME_BETWEEN_DIALOGUE_LINES = 1.3f;
+
+    private NPC currentlyTalkingCharacter;
+    private bool currentlyTyping = false;
+    private bool currentlyWaitingAfterTyped = false;
+    private bool skipWaitingAfterTyped = false;
+    private bool playerInTrigger = false;
+
+    public bool cutsceneStarted { get; private set; } = false;
+    public bool cutsceneFinished { get; private set; } = false;
+
+    private void OnTriggerEnter2D(Collider2D other)
+    {
+        if (other.tag != "Player")
+        {
+            return;
+        }
+
+        playerInTrigger = true;
+
+        if (!cutsceneFinished)
+        {
+            AllowPlayerInteraction(true);
+            /*
+            foreach (NPC character in cutsceneCharacters)
+            {
+                character.OnDialogueTriggerEnter();
+            }
+            */
+        }
+
+        if (!cutsceneStarted)
+        {
+            StartCutScene();
+        }
+    }
+
+    private void OnTriggerExit2D(Collider2D other)
+    {
+        if (other.tag != "Player")
+        {
+            return;
+        }
+
+        playerInTrigger = false;
+
+        if (!cutsceneFinished)
+        {
+            AllowPlayerInteraction(false);
+            /*
+            foreach (NPC character in cutsceneCharacters)
+            {
+                character.OnDialogueTriggerExit();
+            }
+            */
+        }
+    }
+
+    private void AllowPlayerInteraction(bool allowed)
+    {
+        if (allowed)
+        {
+            Player.GetPlayerAction().AddInteractable(this);
+        }
+        else
+        {
+            Player.GetPlayerAction().RemoveInteractable(this);
+        }
+    }
+
+    public bool Interact()
+    {
+        TrySkipCurrentDialogue();
+
+        return true;
+    }
+
+    private void TrySkipCurrentDialogue()
+    {
+        if (currentlyTalkingCharacter != null)
+        {
+            if (currentlyTyping)
+            {
+                currentlyTalkingCharacter.SkipText();
+            }
+            else if (currentlyWaitingAfterTyped)
+            {
+                skipWaitingAfterTyped = true;
+            }
+        }
+    }
+
+    private void StartCutScene()
+    {
+        foreach (NPC character in cutsceneCharacters)
+        {
+            if (character.IsDialogueBoxActive())
+            {
+                character.DeactivateDialogueBox();
+            }
```

Comment:
i'm not sure how reliable this method is to be honest so its prob a good thing to keep in mind for the future

Reply:
yeah i wasn't sure either, i looked around a bunch to see if there was a more definitely method but couldnt find one